### PR TITLE
Feature/38 add user invite functionality rebase

### PR DESCRIPTION
**Pull request by jamiebuckley**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
   resources :contacts
+  resources :users
```
Comment (AlexVPopov) [inline config/routes.rb]: I'd scope this to:

```suggestion
  resources :users, only: [:index, :new, :create]
```

File: db/migrate/20200328103438_add_attributes_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class AddAttributesToUsers < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :email, :text, null: false
```
Comment (AlexVPopov) [inline db/migrate/20200328103438_add_attributes_to_users.rb]: I'd suggest a unique index on the `email`column.

File: app/models/user.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class User < ApplicationRecord
-  belongs_to :organisation
+  belongs_to :organisation, optional: true
```
Comment (AlexVPopov) [inline app/models/user.rb]: Also, please add these changes (optional belongs_to, removed validations) to the user specs as well.

File: db/migrate/20200328103438_add_attributes_to_users.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 class AddAttributesToUsers < ActiveRecord::Migration[6.0]
   def change
     add_column :users, :email, :text, null: false
+    add_index :users, :email
```
Comment (AlexVPopov) [inline db/migrate/20200328103438_add_attributes_to_users.rb]: Perfect, I'd only add `unique: true` as well.

File: spec/factories/users.rb [hunk 5]
```diff
@@ -4,5 +4,7 @@
 
     first_name { Faker::Name.first_name }
     last_name { Faker::Name.last_name }
+    email { Faker::Internet.email }
```
Comment (AlexVPopov) [inline spec/factories/users.rb]: You can add uniqueness here as well: `Faker::Internet.unique.email` (from https://github.com/faker-ruby/faker#ensuring-unique-values).

File: db/migrate/20200328103438_add_attributes_to_users.rb (hunk 4 above)
Comment (jamiebuckley) [reply db/migrate/20200328103438_add_attributes_to_users.rb]: no probs, will add that too

**Pull request merged**